pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2103,10 +2103,22 @@ func (s *AWSCloud) UpdateTCPLoadBalancer(name, region string, hosts []string) er
 	return nil
 }
 
-// Returns the instance with the specified ID
-func (a *AWSCloud) getInstanceById(instanceID string) (*ec2.Instance, error) {
+// Returns the instance with the specified ID - preserving this incase someone wants to use it
+func (a *AWSCloud) getInstanceByID(instanceID string) (*ec2.Instance, error) {
+	instances, err := a.getInstancesByIDs([]*string{&instanceID})
+	if err != nil {
+		return nil, err
+	}
+	if len(instances) > 1 {
+		return nil, fmt.Errorf("multiple instances found for instance: %s", instanceID)
+	}
+
+	return instances[instanceID], nil
+}
+
+func (a *AWSCloud) getInstancesByIDs(instanceIDs []*string) (map[string]*ec2.Instance, error) {
 	request := &ec2.DescribeInstancesInput{
-		InstanceIds: []*string{&instanceID},
+		InstanceIds: instanceIDs,
 	}
 
 	instances, err := a.ec2.DescribeInstances(request)
```

Comment:
Do you have a link for this?  Just want to double-check the error handling semantics around it (in particular, to make sure it doesn't interact badly with paging).

I don't think it's a problem for us, but ideally we would put a link to the docs in there, and we check that an error is returned and we don't e.g. get into a loop repeatedly calling AWS!